### DISCUSSION TITLE
Are addresses (esp house numbers) included in protomaps hosted?

### DISCUSSION THREAD

**Discussion by pietervdvn** [protomaps/basemaps#280] [Q&A] [answered]
Hi,

I'd like to show housenumbers when zoomed in (e.g. z>=16) in MapComplete.

Are housenumbers in the protomaps tiles? If not, are there any plans in including them?

Currently, I don't see them mentioned in [the docs](https://docs.protomaps.com/basemaps/layers), so I assume they are not included.

**Comment by bdon** [comment, marked as answer]
No, not currently. 

I think we should add them both from OSM nodes and also derive a centroid point from buildings with an `addr:` key.

**Reply by do-me** [reply]
That would be amazing and I think it would even make sense on a zoom level 13 - 15 so users can much more quickly navigate between buildings!

**Comment by do-me** [comment]
Hi @bdon, 
I was just trying to create minimal Monaco example with housenumbers. @wipfli was so kind to point me to the file I need to change (https://github.com/protomaps/PMTiles/issues/503). 

- I compiled the `tiles/target/protomaps-basemap-HEAD-with-deps.jar` and ran `java -jar target/*-with-deps.jar --download --force --area=monaco`
- This is the output file: [monaco.pmtiles.zip](https://github.com/user-attachments/files/18040328/monaco.pmtiles.zip)
- If I inspect the tiles with https://pmtiles.io/#map=17.47/43.737039/7.423197, it seems to have worked: 

![image](https://github.com/user-attachments/assets/d5e3a56f-afa4-43ec-b2fc-df44659f3a99)

Now, how could I modify the style to display these housnumbers best? 

Also, housenumbers are not only attached to buildings but also other entities as explained here: https://taginfo.openstreetmap.org/keys/addr:housenumber#overview, so not only `Buildings.java` but other files should be modified too.

**Reply by wipfli** [reply]
> Now, how could I modify the style to display these housnumbers best?

I made a small basemap starter repo at https://github.com/wipfli/basemap-starter. Clone that repo and replace zurich.pmtiles with your own file. Then add something in the style.json file to display the house numbers.

**Reply by wipfli** [reply]
maybe something like this layer:

```json
{
  "id": "housenumbers",
  "source": "protomaps",
  "source-layer": "buildings",
  "layout": {
    "text-field": ["get", "house_number"],
    "text-font": ["Noto Sans Regular"]
  }
}
```

**Reply by do-me** [reply]
Thanks a lot! Will find some time on Wednesday to play with it and report back.

**Reply by nvkelso** [reply]
House numbers should only appear at max tile zoom (15) else they bloat tile size when they generally aren't displayed until zoom 17+.

**Reply by do-me** [reply]
Thanks to @wipfli it worked beautifully! Now it's just a matter of identifying all objects where housnumbers are tagged and fine-tuning the style like @nvkelso proposed. 

Here is the repo: https://github.com/do-me/basemap-starter
Demo: https://do-me.github.io/basemap-starter/#map=14.19/43.73707/7.42386

![image](https://github.com/user-attachments/assets/2072a953-4530-445c-b455-719fc848acc1)

**Comment by bdon** [comment]
The latest hosted version includes `kind=address` in the `buildings` layer, as long as you are using [@protomaps/basemaps](https://www.npmjs.com/package/@protomaps/basemaps) v5.0+.

**Comment by pietervdvn** [comment]
Also see #336 , #380 , #335 .

I've been updating the MapComplete-sunny layer to the latest version (which went well) and added housenumbers, but it seems that only housenumbers on points are included - all the buildings below have numbers:

![image](https://github.com/user-attachments/assets/f985f111-d80a-4554-a13d-84337f0bd78f)

I think there is a little bug in the processing, see [my comment in #380](https://github.com/protomaps/basemaps/pull/380#issuecomment-2674516972)
